File: crates/rr-core/tests/fixtures/java/surface.java
package app;

public class Service implements Runner {
    private int secret;
    int packageField;
    protected int inherited;
    public static final int MAX = 3;

    public Service(String name) {}

    public void run() {
        helper();
    }

    private void hidden() {}

    void packageRun() {}

    @Test
    public void aTest() {}
}

interface Runner {
    void run();
}

enum Mode {
    FAST,
    SLOW
}

public record Point(int x, int y) {}

File: crates/rr-core/tests/fixtures/swift/surface.swift
import Foundation

public class Service {
    public var name: String
    package var group: String
    private var secret: Int

    public init(name: String) {
        self.name = name
        self.group = "x"
        self.secret = 1
    }

    public func run() {
        helper(name)
    }
}

protocol Runner {
    func run()
}

enum Mode {
    case fast, slow
}

struct Point {
    var x: Int
    var y: Int
}

public func bare(_ value: Int) -> Int {
    return value
}
